### PR TITLE
StorageService

### PR DESCRIPTION
**Pull request by jhcastanod**

### CODE REVIEW

File: src/app/services/storage/storage.service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  constructor(private readonly cookieService: CookieService) { }
+
+  get(value: string): string {
+    if (this.localStorage) {
+      const getValue = localStorage.getItem(value);
+
+      return getValue;
+    }
+
+    const getCookie = this.cookieService.get(value);
```
Comment (luqezman) [inline src/app/services/storage/storage.service.ts]: change ```getCookie to cookie```

File: src/app/services/storage/storage.service.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    return getCookie;
```
Comment (luqezman) [inline src/app/services/storage/storage.service.ts]: change ```getCookie to cookie```

File: src/app/services/storage/storage.service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  providedIn: 'root'
+})
+export class StorageService {
+  localStorage = window.localStorage;
+
+  constructor(private readonly cookieService: CookieService) { }
+
+  get(value: string): string {
+    if (this.localStorage) {
+      const getValue = localStorage.getItem(value);
```
Comment (luqezman) [inline src/app/services/storage/storage.service.ts]: change ```getValue to cookie```

File: src/app/services/storage/storage.service.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+      return getValue;
```
Comment (luqezman) [inline src/app/services/storage/storage.service.ts]: change ```getValue to cookie```

File: src/app/services/storage/storage.service.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+import { Storage } from 'interfaces/storage';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StorageService {
+  localStorage = window.localStorage;
+
+  constructor(private readonly cookieService: CookieService) { }
```
Comment (luqezman) [inline src/app/services/storage/storage.service.ts]: Remove space inside `{ }`

File: src/app/services/storage/storage.service.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { CookieService } from 'ngx-cookie-service';
+
+import { Storage } from 'interfaces/storage';
```
Comment (luqezman) [inline src/app/services/storage/storage.service.ts]: This is interface is not needed anymore

File: src/app/app.component.spec.ts [hunk 7]
```diff
@@ -1,6 +1,8 @@
 import { TestBed, async } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
+
 import { AppComponent } from './app.component';
+import { StorageMockClass } from 'providers/storage/strategies/__mocks__/storage.provider';
```
Comment (luqezman) [inline src/app/app.component.spec.ts]: Absolute imports should come before relative imports

File: src/app/app.component.ts [hunk 8]
```diff
@@ -1,10 +1,19 @@
-import { Component } from '@angular/core';
+import { Component, Inject, OnInit } from '@angular/core';
+import { IStorage } from 'providers/storage/interfaces/storage';
```
Comment (luqezman) [inline src/app/app.component.ts]: Add blank line here

File: src/app/app.module.ts [hunk 9]
```diff
@@ -3,6 +3,7 @@ import { NgModule } from '@angular/core';
 
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { StorageProvider } from 'providers/storage/storage.provider';
```
Comment (luqezman) [inline src/app/app.module.ts]: Absolute imports should come before relative imports

File: src/app/providers/storage/storage.provider.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { LocalStorage } from './strategies/local-storage.strategy';
+import { CookieStorage } from './strategies/cookie-storage.strategy';
```
Comment (luqezman) [inline src/app/providers/storage/storage.provider.ts]: Import alphabetically

File: src/app/providers/storage/strategies/__mocks__/storage.provider.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+export class StorageMockClass {
+  get() { }
```
Comment (luqezman) [inline src/app/providers/storage/strategies/__mocks__/storage.provider.ts]: return something dummy

File: src/app/providers/storage/strategies/__mocks__/storage.provider.ts [hunk 12]
```diff
@@ -0,0 +1,5 @@
...
+
+  set() { }
```
Comment (luqezman) [inline src/app/providers/storage/strategies/__mocks__/storage.provider.ts]: return something dummy

File: src/app/providers/storage/strategies/cookie-storage/cookie-storage.strategie.spec.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+      providers: [CookieStorage]
+    });
+
+    strategy = TestBed.get(CookieStorage);
+  });
+
+  it('should create', () => {
+    expect('d').toBeTruthy();
+  });
+
```
Comment (luqezman) [inline src/app/providers/storage/strategies/cookie-storage/cookie-storage.strategie.spec.ts]: Remove extra space

File: src/app/providers/storage/strategies/cookie-storage/cookie-storage.strategie.spec.ts [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (luqezman) [inline src/app/providers/storage/strategies/cookie-storage/cookie-storage.strategie.spec.ts]: Remove extra space

File: src/app/app.module.ts [hunk 15]
```diff
@@ -3,6 +3,7 @@ import { NgModule } from '@angular/core';
...
 
 @NgModule({
   declarations: [
```
Comment (luqezman) [inline src/app/app.module.ts]: sort props alphabetically

File: src/app/providers/storage/strategies/__mocks__/storage.provider.ts (hunk 12 above)
Comment (jhcastanod) [reply src/app/providers/storage/strategies/__mocks__/storage.provider.ts]: What should I return in set method?? In fact, I'm no returning anything

Comment (luqezman) [reply src/app/providers/storage/strategies/__mocks__/storage.provider.ts]: nothing, really sorry about that

**Pull request merged**